src/mbgl/style/paint_property_binder.hpp

```diff
@@ -9,7 +9,22 @@ namespace mbgl {
 namespace style {
 
 template <class T, class A>
-class ConstantPaintPropertyBinder {
+class PaintPropertyBinder {
+public:
+    using BaseAttribute = A;
+    using Attribute = attributes::ZoomInterpolatedAttribute<BaseAttribute>;
+    using AttributeBinding = typename Attribute::Binding;
+
+    virtual ~PaintPropertyBinder() = default;
+
+    virtual void populateVertexVector(const GeometryTileFeature& feature, std::size_t length) = 0;
+    virtual void upload(gl::Context& context) = 0;
+    virtual AttributeBinding attributeBinding(const PossiblyEvaluatedPropertyValue<T>& currentValue) const = 0;
+    virtual float interpolationFactor(float currentZoom) const = 0;
```

Comment:
I don't think so, but in practice the overhead seems to be dwarfed by other parts of rendering; e.g. in a quick profile I see 2ms attributed to `PaintPropertyBinders<...>::attributeBindings` and 148ms to `gl::Program::draw`.